ide/static/js/content.js

```diff
@@ -104,13 +106,25 @@ class Content extends React.Component {
     this.onSocketError = this.onSocketError.bind(this);
     this.waitForConnection = this.waitForConnection.bind(this);
     this.setUserId = this.setUserId.bind(this);
+    this.getUserId = this.getUserId.bind(this);
+    this.getUserName = this.getUserName.bind(this);
+    this.setUserName = this.setUserName.bind(this);
     this.modalContent = null;
     this.modalHeader = null;
     // Might need to improve the logic of clickEvent
     this.clickEvent = false;
     this.handleClick = this.handleClick.bind(this);
     this.performSharedUpdate = this.performSharedUpdate.bind(this);
+    this.performSharedAdd = this.performSharedAdd.bind(this);
+    this.performSharedDelete = this.performSharedDelete.bind(this);
+    this.addHighlightOnLayer = this.addHighlightOnLayer.bind(this);
+    this.addSharedComment = this.addSharedComment.bind(this);
     this.changeCommentOnLayer = this.changeCommentOnLayer.bind(this);
+    this.getRandomColor = this.getRandomColor.bind(this);
+  }
+  getRandomColor() {
```

Comment:
Instead of random colours, can we choose from a colour palette to ensure the colours are different? Because in some cases they are very similar. We can have a set of say 10 colours and then reuse them if there are more users.